### DISCUSSION TITLE
4.4.1: client not receiving events upon connection

### DISCUSSION THREAD

**Discussion by please-fill-out-this-field** [socketio/socket.io#4278] [Q&A] [answered]
**Server code:**

```js
const express = require('express')
const http = require('http')
const { Server } = require('socket.io')

const APP_HOST = process.env.APP_HOST || '192.168.0.100'
const APP_PORT = process.env.APP_PORT || 8081

const app = express()
const server = http.createServer(app)
const io = new Server(server, {
  cors: {
    origin: "*",
    methods: ["PUT", "GET", "POST", "DELETE", "OPTIONS"],
  }
})

io.on('connect', socket => {
  console.log('New connection ' + socket.id)
  setTimeout(() => socket.emit('message', 'well hello there'), 500) // STATEMENT A
  socket.emit('message', 'well hello there') // STATEMENT B
})

server.listen(APP_PORT, APP_HOST, () => {
  console.log(`Listening on ${APP_HOST}:${APP_PORT}`)
})
```

**Client code:**

```html
<!DOCTYPE html>
<html>
  <head>
    <script src="https://cdn.jsdelivr.net/npm/socket.io-client@4.4.1/dist/socket.io.js"></script>
    <script type="text/javascript">document.addEventListener('DOMContentLoaded', function () { var socket = io('ws://192.168.0.100:8081'); socket.connect(); });</script>
  </head>
  <body></body>
</html>
```

**Problem:**

Statement A seems to "work" somewhat - at least client receives the message. Statement B, however, does NOT work at all - client receives nothing but "2probe" and stuff, although it SHOULD work, according to official examples ([here](https://socket.io/get-started/private-messaging-part-2/#persistent-session-id), see the second block).

Obviously, I want this to work without dirty timeout hacks.

Absolutely no idea what might be the issue. Any help is appreciated.

**Comment by darrachequesne** [comment]
Hi! I can't reproduce: https://github.com/socketio/socket.io-fiddle/tree/issues/socket.io/4278

![Screenshot from 2022-02-12 23-23-48](https://user-images.githubusercontent.com/13031701/153730437-f3fbff32-a9b3-4c6d-bfb6-7ea1c4738f68.png)

Could you please check?

**Reply by please-fill-out-this-field** [reply]
@darrachequesne Thanks. Checked.

**Comment by please-fill-out-this-field** [comment, marked as answer]
Solved.
Client actually receives both messages, but first one never shows up in Firefox network debugger for some reason.
It appears to be a Firefox issue.
